app/Console/Command/FindOutOfSyncProjections.php

```diff
@@ -0,0 +1,123 @@
+
+final class FindOutOfSyncProjections extends Command
+{
+    private Connection $connection;
+    private DocumentRepository $eventDocumentRepository;
+    private DocumentRepository $placeDocumentRepository;
+    private DocumentRepository $organizerDocumentRepository;
+
+    public function __construct(
+        Connection $connection,
+        DocumentRepository $eventDocumentRepository,
+        DocumentRepository $placeDocumentRepository,
+        DocumentRepository $organizerDocumentRepository
+    ) {
+        $this->connection = $connection;
+        $this->eventDocumentRepository = $eventDocumentRepository;
+        $this->placeDocumentRepository = $placeDocumentRepository;
+        $this->organizerDocumentRepository = $organizerDocumentRepository;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $aggregateTypeEnumeration = implode(
+            ', ',
+            AggregateType::getAllowedValues()
+        );
+
+        $this
+            ->setName('find-out-of-sync-projections')
+            ->setDescription('Find projections that are behind with the events inside the event store')
+            ->addArgument(
+                'aggregate-type',
+                InputArgument::REQUIRED,
+                'Aggregate type to find missing projections. One of: ' . $aggregateTypeEnumeration . '.',
+            )
+            ->addArgument(
+                'first-id',
+                InputArgument::REQUIRED,
+                'Id of the first row to process.',
+            )
+            ->addArgument(
+                'last-id',
+                InputArgument::REQUIRED,
+                'Id of the last row to process.',
+            );;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $aggregateType = new AggregateType($input->getArgument('aggregate-type'));
+
+        $results = $this->connection->createQueryBuilder()
+            ->select('uuid', 'MAX(recorded_on) as recorded')
+            ->from('event_store')
+            ->where('aggregate_type = :aggregate_type')
+            ->setParameter(':aggregate_type', $aggregateType->toString())
+            ->andWhere('id >= :first_id')
+            ->setParameter(':first_id', $input->getArgument('first-id'))
+            ->andWhere('id <= :last_id')
+            ->setParameter(':last_id', $input->getArgument('last-id'))
+            ->groupBy('uuid')
+            ->execute()
+            ->fetchAll();
+
+        foreach ($results as $result) {
+            $uuid = $result['uuid'];
+            $recorded = $result['recorded'];
+            $modified = $this->getModified($uuid, $aggregateType, $output);
+
+            if ($modified) {
+                $recordedDate = new DateTime($recorded);
+                $modified = new DateTime($modified);
+
+                $difference = $modified->diff($recordedDate);
+                if ($difference->s > 5) {
```

Comment:
The `modified` is based on the last `recorded_on`, so it should be exactly the same in theory.

- The code for offers: https://github.com/cultuurnet/udb3-backend/blob/7f01c6d10e34612b08041b5ed13bcaabcc2d41bf/src/Offer/ReadModel/JSONLD/OfferLDProjector.php#L839
- The code for organizers: https://github.com/cultuurnet/udb3-backend/blob/0f5004011dc77b74afa500bf27780b6e170e5566/src/Organizer/OrganizerLDProjector.php#L655
- And for example in a test assertion: https://github.com/cultuurnet/udb3-backend/blob/9b1b6befd85083ea4860ddea9a5b50d0f7e864be/tests/OfferLDProjectorTestBase.php#L128

Reply:
Maybe we check for a 1-second difference because the `recorded_on` inside the database has milliseconds and the `modified` inside the json projection only seconds?

Reply:
You can convert the `recorded_on` to the exact value that it should be in `modified` using `RecordedOn::toString()` and compare that they are exactly the same or not? We don't need to know the date/time difference, just that they match or not.